apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -1342,21 +1377,265 @@ private void StartPush()
                     _pushInFlight += acks.Count;
                     _concurrentPushCount += 1;
                     Task ackTask = _client.AcknowledgeAsync(_subscriptionName, acks, _hardStopCts.Token);
-                    Add(ackTask, Next(false, () => HandleAckResponse(ackTask, acks, null, null)));
+                    Add(ackTask, Next(false, _exactlyOnceDeliveryEnabled ? () => HandleAckResponseForExactlyOnce(ackTask, acks, null, null) : () => HandleAckResponse(ackTask, acks, null, null)));
```

Comment:
Two options for avoiding the redundancy here:

1. Declare `Action<Task, List<string>, List<string>, List<TimedId>> ackHandler = _exactlyOnceDeliveryEnabled ? HandleAckResponseForExactlyOnce : HandleAckResponse;` just above this (line 1375), and then use `ackHandler` here and in the two uses below.
2. Just change `HandleAckResponse` to split off into HandleAckResponseForExactlyOnce if _exactlyOnceDeliveryEnabled is true. Note that this would (I guess) potentially act differently as the choice about which path is taken would be made after the acknowledgement has been received rather than here. I don't know whether that matters.

Reply:
Implemented the second option.